### ISSUE TITLE
Lifetimes for structs and traits in inlay hints don't work in VSCode

### ISSUE THREAD

**Issue by coolCucumber-cat** [rust-lang/rust-analyzer#17098]
**rust-analyzer version**: 0.3.1924-standalone (5dbe3fe75 2024-04-14)

It also only occurs on this version (0.3.1924) and not previous ones like 0.3.1916

**rustc version**: rustc 1.77.2 (25ef9e3d8 2024-04-09)

**editor or extension**: VSCode

**code snippet to reproduce**: 
```rust
struct Test<'a> {
	t: &'a i32,
}

fn main() {
	let test = Test { t: &0 };
}
```

The inlay hint for test is `Test<'{error}>`. You can also not turn them off. I don't want any lifetimes in inlay hints at all, they take up too much space, so an option to turn them off would be a win-win.

**Comment by Veykril** [comment]
Right, we should not show error lifetimes for now as we have too many of them due to missing implementations

**Comment by coolCucumber-cat** [comment]
@Veykril I'm not sure what you mean? Do you mean lifetime errors instead of error lifetimes?

**Comment by Veykril** [comment]
I'm saying we shouldn't render `'{error}` in inlay hints currently (or generally), as most lifetimes will end up as errors currently

**Comment by Leedehai** [comment]
Adding a Settings option to disable showing lifetime specifiers in the inlay hints (`'{error}` or otherwise) would be great.
